First baseline results added, rewrite experiments, remove unused sections (#49)
chhoumann/thesis-chemcam: report_pre_thesis/src/sections/experiments.tex

\section{Experiments}\label{sec:experiments}
We describe our replica of the MOC pipeline, compare it to the original, and describe the experiments we have conducted.
The performance of our replica serves as baseline for comparison with the original MOC pipeline and as a basis for assessing experimental results.

\subsection{Replication of the MOC Pipeline}\label{sec:replica_moc}
We present the baseline RMSEs of the original and our replicas of the PLS1-SM, ICA, and MOC models in Table~\ref{tab:results_rmses}.
Figure~\ref{fig:rmse_histograms} illustrates the distribution of the RMSEs of the original and our replicas of the PLS1-SM, ICA, and MOC models as a grouped histogram.
The results show that the RMSEs of our replicas of the PLS1-SM, ICA and MOC models are similar to the original models.
However, the are some notable differences --- in some cases, our replicas outperform the original models, while in other cases, the original models outperform our replicas.
These differences can be attributed to a number of factors.

> Firstly, the original pipeline was trained on two datasets, one acquired at a 1600mm standoff distance and one acquired at a 3000mm standoff distance.

Reviewer: Models fit better here

We have only used the 1600mm dataset for our replicas since we do not have access to the 3000mm dataset.
As mentioned in Section~\ref{sec:outlier_removal}, we also chose chose to automate the outlier removal process for the PLS1-SM phase, whereas the original authors performed this manually.
Moreover, we chose to exclude the outlier removal step during the ICA phase to avoid introducing unsubstantiated assumptions, as described in Section~\ref{sec:ica_data_preprocessing}.

> As we also mentioned in Section~\ref{sec:ica_data_preprocessing}, we only used one of the five location datasets for each sample, whereas the original authors used all five due to missing information about how these datasets were used.

Reviewer: ```suggestion
In Section~\ref{sec:ica_data_preprocessing}, we clarified that unlike the original authors who utilized all five location datasets, our analysis was limited to a single dataset per sample due to the absence of details on their integration.
```

\ce{TiO2}  & 0.94               & 0.47              & 1.44           & 0.63          & 1.03           & 0.49 \\
\ce{Al2O3} & 2.85               & 1.94              & 4.77           & 5.55          & 3.47           & 2.39 \\
\ce{FeO_T} & 2.01               & 4.35              & 5.17           & 8.30          & 2.31           & 5.21 \\
\ce{MgO}   & 1.06               & 1.17              & 4.08           & 2.90          & 2.21           & 1.67 \\
\ce{CaO}   & 2.65               & 1.43              & 3.07           & 3.52          & 2.72           & 1.81 \\
\ce{Na2O}  & 0.62               & 0.66              & 2.29           & 1.72          & 0.62           & 1.10 \\
\ce{K2O}   & 0.72               & 0.72              & 0.98           & 1.37          & 0.82           & 1.09 \\
\hline
\end{tabular*}
\caption{RMSE of the original and our replicas of the PLS1-SM, ICA, and MOC models.}
\label{tab:results_rmses}
\end{table*}

\begin{figure*}[b]
	\centering
	\includegraphics[width=0.85\textwidth]{images/rmse_historgram.png}
	\caption{Grouped histogram of the RMSEs of the original and our replicas of the PLS1-SM, ICA, and MOC models.}
	\label{fig:rmse_histograms}
\end{figure*}

> To evaluate the statistical significance of the differences between the original and our replicated models (PLS1-SM, ICA, and MOC), we applied three statistical tests: Student's t-test, Welch's t-test, and the Wilcoxon signed-rank test.

Reviewer: Perhaps we examine this deeper, select the best one (of these, or another, or all three if we can prove they make sense), and give the formula & give explanation for why we use that particular test. Then we explain results of the single test.

\begin{enumerate}
	\item\label{enum:experiment1} No outlier removal for PLS1-SM to evaluate the significance of our automated outlier removal process and whether outlier removal is necessary.
	\item\label{enum:experiment2} Maintain the threshold values from the second iteration of the outlier removal in the PLS1-SM training process to see if a more or less conservative approach is advantageous.
	\item\label{enum:experiment3} MAD for ICA to evaluate the significance of the outlier removal step in the ICA phase.
	\item\label{enum:experiment4} Evaluating the impact on ICA model performance when using all five location datasets versus just one.
	\item\label{enum:experiment5} Other models to compare the performance of the PLS1-SM and ICA models to other models.
	% \item PLS outlier removal on ICA (?) instead of MAD.
\end{enumerate}

\noindent
We have chosen these experiments based on the following reasons.
For Experiment~\ref{enum:experiment1}, we are interested in evaluating the significance of our automated outlier removal process due to its differences from the original manual process.

> Additionally, we are interested in examining whether outlier removal is necessary for the PLS1-SM phase by comparing the results with and without outlier removal.

Reviewer 2: I think perhaps this is a less interesting reason. One could argue that the original authors did outlier removal because they found it necessary.

I think we should instead focus on trying to figure out what impact our outlier detection has on the results and hence why it makes sense to do a test without outlier removal. Essentially, I would like to just see more of the topic from the first paragraph.

```suggestion
By conducting an experiment without outlier detection we will be able to discern the benefit of our outlier removal approach.
Additionally, this experiment could also yield insights into the benefit of the original, manual outlier detection approach and help determine whether an automated approach can lead to better predictions.
```

Something like that.

Reviewer: We do both

For Experiment~\ref{enum:experiment2}, we want to evaluate the significance of the threshold values used in the outlier removal process by maintaining the threshold values from the second iteration of the outlier removal in the PLS1-SM training process.
This is a more conservative than always updating the threshold values after each iteration.

For Experiment~\ref{enum:experiment3}, we implement our best guess of the original authors' outlier removal process for the ICA phase, which is based on the Median Absolute Deviation (MAD).
Then, we compare the results with and without MAD to evaluate the significance of the outlier removal step in the ICA phase.

For Experiment~\ref{enum:experiment4}, we perform ICA using all five location datasets by averaging the results from location into a single dataset.
By doing this, we will either obtain a more representative dataset for the sample, or we will end up losing information by averaging the results.

> For Experiment~\ref{enum:experiment5}, we want to compare the performance of the MOC model with other models.

Reviewer 2: Perhaps also a little more justification here.

Specifically, we train an XGBoost model and an ANN model for comparison.

> 

Reviewer 2: I just read the report in the PDF and I think my confusion about the justifications comes from the fact that there is no clear connection between the "chosen experiments" section and the following experiment sections where we outline the reasons a little more. I think we need to work on tying these together a little more.

	\label{fig:rmse_histograms}
\end{figure*}

> \citet{andersonImprovedAccuracyQuantitative2017} used the Student's t-test to show their new model outperformed the old one. In our study, we apply the same test but with a different aim: to verify if our replicated models (PLS1-SM, ICA, and MOC) are statistically similar to the original MOC  models, presented in \citet{cleggRecalibrationMarsScience2017}. This approach allows us to assess whether our models demonstrate equivalence rather than improvement.

Author: ```suggestion
\citet{andersonImprovedAccuracyQuantitative2017} used the Student's t-test to show their new model outperformed the old one. In our study, we apply the same test but with a different aim: to verify if there is no statistical significant difference between our replicated models (PLS1-SM, ICA, and MOC) and the original models presented in \citet{cleggRecalibrationMarsScience2017}. This approach allows us to assess whether our models demonstrate equivalence rather than improvement.
```

$$
where $n$ is the number of samples, and $\Gamma$ is the gamma function. This formulation captures the variance of the RMSE, reflecting the dispersion of error magnitudes.
Subsequently, the t-statistic $t$ is calculated to evaluate the statistical significance of the difference between the RMSEs of the original and replicated models:
$$
t = \frac{\text{RMSE}_A - \text{RMSE}_B}{\sqrt{S_{\text{RMSE}_A}^2 + S_{\text{RMSE}_B}^2}}
$$
where $\text{RMSE}_A$ and $\text{RMSE}_B$ correspond to the RMSE values of the original and replicated models, respectively.
The degrees of freedom ($f$) associated with this comparison are determined by:
$$
f = \frac{\left(S_{\text{RMSE}_A}^2 + S_{\text{RMSE}_B}^2\right)^2}{\frac{S^4_{\text{RMSE}_A}}{n_A - 1} + \frac{S^4_{\text{RMSE}_B}}{n_B - 1}}
$$
which accounts for the variances in RMSE values and the respective sample sizes ($n_A$ and $n_B$) of the models being compared.
Finally, the p-value is derived from the t-distribution's cumulative distribution function:
$$p\text{-value} = 2 \times \left(1 - F_{\text{T}}(|t|)\right)$$

> where $F_{\text{T}}$ is the cumulative distribution function of the t-distribution with $f$ degrees of freedom, providing the probability that the observed difference in RMSEs could occur under the null hypothesis.

Author: Not sure I like the phrasing "providing the probability that the observed difference in RMSEs could occur under the null hypothesis".
Maybe something like this?
```suggestion
where $F_{\text{T}}$ is the cumulative distribution function of the t-distribution with $f$ degrees of freedom. 
Higher p-values indicate greater compatibility with the null hypothesis.
```

Reviewer: \> "Maybe something like this"
\> deletes the text

lol

\hline
\ce{SiO2} & 42.00\% & 48.75\% & 38.45\% \\
\ce{TiO2} & 9.73\% & 6.31\% & 8.23\% \\
\ce{Al2O3} & 30.23\% & 67.06\% & 31.58\% \\
\ce{FeO_T} & 7.48\% & 21.68\% & 6.57\% \\
\ce{MgO} & 78.05\% & 35.44\% & 44.08\% \\
\ce{CaO} & 12.73\% & 70.04\% & 27.79\% \\
\ce{Na2O} & 85.96\% & 43.16\% & 14.93\% \\
\ce{K2O} & 100.00\% & 36.27\% & 43.40\% \\
\hline
\end{tabular}
\caption{Results of the PLS1-SM, ICA, and MOC t-tests}
\label{table:results_ttests}
\end{table}

The results from our t-tests are summarized as follows:

\begin{itemize}
    \item \textbf{PLS1-SM:} The p-values observed across the spectrum, most notably the 100.00\% for $K_2O$, unequivocally suggest a strong statistical proximity to the original models for a majority of the parameters. This is particularly indicative of a successful replication effort. However, the relatively lower p-value for $FeO_T$ (7.48\%) underscores an area where the model's performance diverges from the original. Given the high variance noted in the Section~\ref{sec:data_overview} for $FeO_T$ compositions, this deviation could be ascribed to inherent data variability rather than model inaccuracy.

>     \item \textbf{ICA:} This model demonstrates considerable alignment with the original models across several constituents, as evidenced by p-values like 67.06\% for $Al_2O_3$ and 70.04\% for $CaO$. These figures signify a commendable approximation to the original models’ performance. Conversely, the p-value for $TiO_2$ (6.31\%) marks an exception, indicating a region where the ICA model might benefit from refinement. The noted high variance in $TiO_2$ values within our dataset likely contributes to this outlier, pointing to data variability as a contributing factor.

Author: Don't like the word commendable here.
```suggestion
    \item \textbf{ICA:} This model demonstrates considerable alignment with the original models across several constituents, as evidenced by p-values like 67.06\% for $Al_2O_3$ and 70.04\% for $CaO$. These figures indicate a noteworthy approximation to the original models' performance. Conversely, the p-value for $TiO_2$ (6.31\%) marks an exception, indicating a region where the ICA model might benefit from refinement. The noted high variance in $TiO_2$ values within our dataset likely contributes to this outlier, pointing to data variability as a contributing factor.
```

    \item \textbf{MOC:} The replicated MOC model's p-values, especially the 43.40\% for $K_2O$, reiterate its consistency with the original model's outcomes. This aligns with our hypothesis of statistical equivalence. Yet, as with PLS1-SM, the $FeO_T$ component exhibits a lower p-value (6.57\%), highlighting an area of divergence potentially attributed to the previously discussed variance in $FeO_T$ compositions.
\end{itemize}

> Our comparative analysis, enriched by an acknowledgment that higher p-values denote model fidelity, predominantly affirms the statistical equivalence of our replicated models to the original MOC models across most tested parameters.

Author: ```suggestion
This comparative analysis affirms a satisfying statistical equivalence between our replicated models and the original MOC models across most tested parameters.
```

This alignment underscores the robustness of our replication efforts.

However, the lower p-values associated with $FeO_T$ for both PLS1-SM and MOC, in addition to $TiO_2$ for ICA, flag these elements as focal points for potential refinement.
Given our dataset's inherent variability in the compositions of these specific elements, as substantiated in Section~\ref{sec:data_overview}, these findings are interpretable and not wholly unexpected.

> In conclusion, our t-tests demonstrate that our replicated models are statistically similar to the original models, thereby achieving our goal of demonstrating equivalence rather than disparity.

Author: ```suggestion
In conclusion, our t-tests demonstrate that our replicated models are statistically similar to the original models, thereby achieving our goal of demonstrating equivalence rather than disparity.
As such, they serve their purpose as a baseline for identifying which aspects of the pipeline contribute the most to the overall error. 
```
